core/selection.ts

```diff
@@ -73,7 +73,7 @@ class Selection {
             );
           }
           const triggeredByTyping = mutations.some(
-            mutation =>
+            (mutation: MutationRecord) =>
```

Comment:
We should change the signature of the upper level function to give `mutations` a type so that we don't need to add `: MutationRecord` here.